### PR TITLE
better link support. if no proto assume https. url support in text

### PR DESCRIPTION
**Pull request by chrisnojima**
fixes weird links not parsing correctly and links not working
@keybase/react-hackers

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/markdown/parser.pegjs [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+            const end = start + match.length
+			const left = text.substring(0, start)
+            const right = text.substring(start + end)
+			return {
+				type: 'text',
+				children: [
+                	...(left ? [left] : []),
+					{type: 'link', children: [match], href},
+                    ...(right ? [right] : []),
+				]
```
Comment (MarcoPolo) [inline shared/markdown/parser.pegjs]: spacing is weird

Comment (chrisnojima) [inline shared/markdown/parser.pegjs]: yeah its fixed in https://github.com/keybase/client/pull/5501

**Comment by chrisnojima** [comment]
@MarcoPolo please re-review. i changed the logic of the link matches a lot
now we detect and toss all non http/https protocols
i allow a/../b but not a..b or a...b

**Comment by keybase-ci-visdiff** [comment]
The commits 62404b49e102...d9d9769a5655 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211575489-62404b49e102-d9d9769a5655-linux/Markdown-links-62404b49e102.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211575489-62404b49e102-d9d9769a5655-linux/Markdown-links-d9d9769a5655.png)  
   Markdown-links comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 62404b49e102...67b6085070c9 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211804900-62404b49e102-67b6085070c9-linux/Markdown-links-62404b49e102.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211804900-62404b49e102-67b6085070c9-linux/Markdown-links-67b6085070c9.png)  
   Markdown-links comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 62404b49e102...0ebe5dfa017b introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211903861-62404b49e102-0ebe5dfa017b-linux/Markdown-links-62404b49e102.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485211903861-62404b49e102-0ebe5dfa017b-linux/Markdown-links-0ebe5dfa017b.png)  
   Markdown-links comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 35275762a2b6...d7cf59ba9f28 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Markdown-links** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485213514382-35275762a2b6-d7cf59ba9f28-linux/Markdown-links-35275762a2b6.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1485213514382-35275762a2b6-d7cf59ba9f28-linux/Markdown-links-d7cf59ba9f28.png)  
   Markdown-links comparison
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Pull request merged**